pkg/route/validation/validation_test.go

```diff
@@ -1200,6 +1639,123 @@ func TestValidateRouteUpdate(t *testing.T) {
+		{description: "hdr with syntax error", input: `%[XYZ.hdr(Host]`, validInput: false},
+		{description: "hdr with url", input: `%[XYZ.hdr(Host)] http://url/hack`, validInput: true},
+
+		{description: "hdr with valid X-XSS-Protection value", input: `1;mode=block`, validInput: true},
+		{description: "hdr with valid Content-Type value", input: `text/plain,text/html`, validInput: true},
+		{description: "hdr with url", input: `text/plain,text/html`, validInput: true},
+
+		{description: "incomplete expression", input: `%[req`, validInput: false},
+		{description: "quoted field", input: `%[XYZ.hdr(%{+Q}Host)]`, validInput: false},
+		{description: "value with conditional expression", input: `%[XYZ.hdr(Host)] if foo`, validInput: true},
+		{description: "value with what looks like a conditional expression", input: `%[XYZ.hdr(Host)]\ if\ foo`, validInput: true},
+		{description: "unsuported fetcher", input: `%[date(3600),http_date]`, validInput: false},
+		{description: "nonexistent sample fetches or converters", input: `%[foo,lower]`, validInput: false},
+		{description: "nonexistent converters", input: `%[req.hdr(host),foo]`, validInput: false},
+		{description: "missing parentheses or braces", input: `%{Q[req.hdr(host)]`, validInput: false},
+		{description: "missing parentheses or braces", input: `%Q}[req.hdr(host)]`, validInput: false},
+		{description: "missing parentheses or braces", input: `%{{Q}[req.hdr(host)]`, validInput: false},
+		{description: "missing parentheses or braces", input: `%[req.hdr(host)`, validInput: false},
+		{description: "missing parentheses or braces", input: `%req.hdr(host)]`, validInput: false},
+		{description: "missing parentheses or braces", input: `%[req.hdrhost)]`, validInput: false},
+		{description: "missing parentheses or braces", input: `%[req.hdr(host]`, validInput: false},
+		{description: "missing parentheses or braces", input: `%[req.hdr(host`, validInput: false},
+		{description: "missing parentheses or braces", input: `%{req.hdr(host)}`, validInput: false},
+		{description: "parameters for a sample fetch that doesn't take parameters", input: `%[ssl_c_der(host)]`, validInput: false},
+		{description: "dangerous sample fetches and converters", input: `%[env(FOO)]`, validInput: false},
+		{description: "dangerous sample fetches and converters", input: `%[req.hdr(host),debug()]`, validInput: false},
+		{description: "extra comma", input: `%[req.hdr(host),,lower]`, validInput: false},
+
+		// CR and LF are not allowed in header value as per RFC https://datatracker.ietf.org/doc/html/rfc7230#section-3.2.4
+		{description: "linefeed", input: "\r", validInput: false},
+		{description: "CRLF", input: "\r\n", validInput: false},
+
+		// This value is allowed in haproxy.config and does not cause haproxy to crash. The input strings are sanitized by escaping single quotes in the value and then
+		// single quoting the whole value which helps haproxy not to crash in the event of an invalid string provided.
```

Comment:
The comment mentions "single quoting the whole value" but I don't see any quoting here. The comment seems only relevant to the places where we emit a value (into the haproxy.cfg file).

Reply:
yes. just added this for providing info.